added README for fusion_filtering (#376)
AlexsLemonade/OpenPBTA-analysis: analyses/fusion_filtering/README.md

## Fusion prioritization

> This analysis will perform artifact filtering and annotation on fusion calls to prioritize putative oncogenic fusions. Briefly, we considered all inframe and frameshift fusion calls with a minimum of 1 junction reads and at least one gene partner expressed (TPM > 1) to be true calls. If a fusion call had large number of spanning fragment reads compared to junction reads (spanning fragment minus junction read greater than ten), we removed these calls as potential false positives. We prioritized a union of fusion calls as true calls if the fused genes were detected by both callers, the same fusion was recurrent within a broad_histology (>2 samples) or the fusion was specific to the broad_histology. If either 5' or 3' genes fused to more than five different genes within a sample, we removed these calls as potential false positives. We annotated putative driver fusions and prioritized fusions based on partners containing known kinases, oncogenes, tumor suppressors, curated transcription factors, cosmic conses gene list or TCGA fusions.

Reviewer: Can you make this one sentence per line for easier commenting?

Author: Done!

## Fusion prioritization

**Module authors :** Krutika Gaonkar ([@kgaonkar6](https://github.com/kgaonkar6)), Jaclyn Taroni ([@jaclyn-taroni](https://github.com/jaclyn-taroni)), Jo Lynne Harenza([@jharenza](https://github.com/jharenza)), and Komal Rathi ([@komalsrathi](https://github.com/komalsrathi))

This analysis is designed to filter artifacts and annotate fusion calls from STARfusion and Arriba fusion callers with the goal of prioritizing oncogenic fusions. 
We considered all inframe and frameshift fusion calls with a minimum of 1 junction reads and at least one gene partner expressed (FPKM > 1) to be potential true calls. 
We then removed fusion calls that had many spanning fragment reads compared to junction reads (`spanning fragment read count` minus `junction read count` greater than ten) as potential false positives. 

> We prioritized a union of fusion calls as true calls if the fused genes were detected by both callers OR the same fusion was recurrent within a broad_histology (>2 samples) OR the fusion was specific to the broad_histology. 

Reviewer: ```suggestion
We retained fusion calls if the fused genes were detected by both callers, the same fusion was recurrent within a broad_histology (>2 samples), the fusion was specific to the broad_histology. 
```

* pbta-gene-expression-rsem-fpkm.strande.rds : aggregated stranded fpm data

#### Inputs used as reference
* genelistreference.txt : known kinases, oncogenes, tumor suppressors, curated transcription factors [@doi:10.1016/j.cell.2018.01.029], COSMIC census genes. MYBL1 [@doi:10.1073/pnas.1300252110], SNCAIP [@doi:10.1038/nature11327], FOXR2 [@doi:10.1016/j.cell.2016.01.015], TTYH1 [@doi:10.1038/ng.2849], and TERT [@doi:10.1038/ng.3438; @doi:10.1002/gcc.22110; @doi:10.1016/j.canlet.2014.11.057; @doi:10.1007/s11910-017-0722-5] were added to the oncogene list and BCOR [@doi:10.1016/j.cell.2016.01.015] and QKI [@doi:10.1038/ng.3500] were added to the tumor suppressor gene list based on pediatric cancer literature review
* fusionreference.txt :  known TCGA fusions
* Brain_FPKM_hg38_matrix.txt.zip : GTex brain samples FPKM data

#### Outputs saved to data download
* pbta-fusion-putative-oncogenic.tsv
* pbta-fusion-recurrently-fused-genes-byhistology.tsv
* pbta-fusion-recurrently-fused-genes-bysample.tsv

### Run script
`bash run_fusion_merged.sh` 

#### Order of scripts in analysis
`01-fusion-standardization.R` : Standardizes fusion calls from STARFusion and Arriba

> `02-fusion-filtering.R` : Artifact filtering by removing readthroughs and NEIGHBORS from annots column; annots column also contains red flag databases that are used to further filter common/normal occuring fusions; all fusions where both genes are expressed <1TPM is removed as non-expressed fusions; fusions required to have atleast 1 JunctionSpanningRead minus  SpanningFragCount-JunctionReadCount to be <10

Reviewer 2: ```suggestion
`02-fusion-filtering.R` : Artifact filtering by removing readthroughs and NEIGHBORS from annots column; annots column also contains red flag databases that are used to further filter common/normal occuring fusions; all fusions where both genes are expressed FPKM < 1 are removed as non-expressed fusions; fusions required to have at least 1 JunctionSpanningRead minus  SpanningFragCount-JunctionReadCount to be <10
```

> `03-Calc-zscore-annotate.R` : Calculates zscore for gene fused gene's expression compared to GTeX brian samples and annotates if differently expressed or not

Reviewer 2: ```suggestion
`03-Calc-zscore-annotate.R` : Calculates z-score for gene fused gene's expression compared to GTeX brain samples and annotates if differently expressed or not
```

> `04-project-specific-filtering.Rmd` : Notebook to perform project specific filtering. We removed fusions with genes fused more than 5 times in a samples as potential artifact. We kepth fusions that were called by both callers and if >2 samples per histology called the fusion. We then prioritize the fusions as putative-oncogenic fusions if any fused gene in the fusion is annotated as kinases, oncogenes, tumor suppressors, curated transcription factors, cosmic conses gene list or is a TCGA fusions

Reviewer 2: Can you change the wording here to match the changes that you've made above? For example, this says `cosmic conses gene list`. It's not clear to me if it should be COSMIC consensus (line 11) or COSMIC census (line 22).

Author: right I'm updating that, it should be COSMIC Cancer Gene Census in all places.

> `05-recurrent-fusions-per-histology.R` : To identify recurrent fusion and fused genes we first indentified RNAseq samples that can be used independetly for each patient. After the selection of samples we identify which fusions and genes are reccurent ( found in >3 participants per histology ) in our pbta-fusion-putative-oncogenic.tsv dataset.

Reviewer 2: ```suggestion
`05-recurrent-fusions-per-histology.R` : To identify recurrent fusion and fused genes we first identified RNA-seq samples that can be used independently for each patient. After the selection of samples we identify which fusions and genes are recurrent (found in >3 participants per histology) in our `pbta-fusion-putative-oncogenic.tsv` dataset.
```

## Fusion prioritization

> **Module author:** Krutika Gaonkar ([@kgaonkar6](https://github.com/kgaonkar6))

Reviewer: ```suggestion
**Module authors :** Krutika Gaonkar ([@kgaonkar6](https://github.com/kgaonkar6)), Jaclyn Taroni ([@jaclyn-taroni](https://github.com/jaclyn-taroni)), Jo Lynne Harenza([@jharenza](https://github.com/jharenza)), and Komal Rathi ([@komalsrathi](https://github.com/komalsrathi))
```

> Jaclyn Taroni ([@jaclyn-taroni] (https://github.com/jaclyn-taroni))

Reviewer: ```suggestion
```

> Jo Lynne([@jharenza](https://github.com/jharenza))

Reviewer: ```suggestion
```

> Komal Rathi ([@komalsrathi] (https://github.com/komalsrathi))

Reviewer: ```suggestion
```

> This analysis will perform artifact filtering and annotation on fusion calls from STARfusion and Arriba fusion callers to prioritize putative oncogenic fusions. 

Reviewer: ```suggestion
This analysis is designed to filter artifacts and annotate fusion calls from STARfusion and Arriba fusion callers with the goal of prioritizing oncogenic fusions. 
```

> Briefly, we considered all inframe and frameshift fusion calls with a minimum of 1 junction reads and at least one gene partner expressed (TPM > 1) to be true calls. 

Reviewer: ```suggestion
We considered all inframe and frameshift fusion calls with a minimum of 1 junction reads and at least one gene partner expressed (TPM > 1) to be potential true calls. 
```

Reviewer 2: I also think this should be `FPKM > 1` rather than `TPM > 1`?

> If a fusion call had large number of spanning fragment reads compared to junction reads (spanning fragment minus junction read greater than ten), we removed these calls as potential false positives. 

Reviewer: ```suggestion
We then removed fusion calls that had many spanning fragment reads compared to junction reads (`spanning fragment read count` minus `junction read count` greater than ten) as potential false positives. 
```

> We prioritized a union of fusion calls as true calls if the fused genes were detected by both callers OR the same fusion was recurrent within a broad_histology (>2 samples) OR the fusion was specific to the broad_histology. 

Reviewer: What does prioritized mean here? Does it mean did not remove or something else? It's possible that I don't understand the filtering steps/starting point instead of not understanding this.

Author: The starting point for this step is QC filtered fusion calls (readthrough, annotated by FusionAnnotator with red-flag/normal fusion 
 database  and expression based filtering)
Now in this step we want to do project specific filtering where the aim is to **keep** fusions that fit the conditions : detected by both callers (which would be our high confidence calls) OR the same fusion was recurrent within a broad_histology (>2 samples to find recurrent fusions per histology) OR the fusion was specific to the broad_histology (that is only seen in this specific histology ) as FilteredFusion.tsv output file.

> If either 5' or 3' genes fused to more than five different genes within a sample, we removed these calls as potential false positives. 

Reviewer: ```suggestion
We removed calls for which one gene was 5' or 3' fused to more than five different other genes within a sample as potential false positives. 
```

> We annotated putative driver fusions and prioritized fusions based on partners containing known kinases, oncogenes, tumor suppressors, curated transcription factors, cosmic conses gene list or TCGA fusions.

Reviewer: ```suggestion
We annotated putative driver fusions and prioritized fusions when at least one fused gene was a known kinase, oncogene, tumor suppressor, curated transcription factor, on the cosmic consensus gene list.
We also annotated fusions between pairs of genes that were observed in TCGA.
```
I think this means the same thing as what you wrote but I was really not sure on the TCGA bit. I didn't know if one member had to be in the TCGA fusion or if both did. This version specifies that both do.

Author: you are right, the fusion between the pair of genes needs to be in TCGA
